db.go

```diff
@@ -55,6 +55,12 @@ type DB struct {
 	// THIS IS UNSAFE. PLEASE USE WITH CAUTION.
 	NoSync bool
 
+	// When true, skips the truncate call when resizing the database.
+	// Setting this to true is only safe on non-ext3/ext4 systems.
+	//
```

Comment:
I wasn't able to determine the root cause. Truncation did work very quickly in some cases but in others it was taking >10m. It may have been a configuration issue or a hardware issue so I'd rather not assign blame to the FS.

Also, `NoTruncate` is nice if you don't want to preallocate space so I'll list that as a benefit.